Fix build failure with MacPorts clang 9 and later:

clang: error: '-I-' not supported, please use -iquote instead

https://gitlab.com/bogofilter/bogofilter/issues/1
--- configure.orig	2019-10-11 15:24:49.000000000 -0500
+++ configure	2019-11-26 02:32:37.000000000 -0600
@@ -6465,7 +6465,7 @@
 	4.*) gcc4=yes ;;
     esac
     case "$CC" in
-	*clang) gcc4=yes ;;
+	*clang*) gcc4=yes ;;
     esac
 fi
  if test $gcc4 = yes; then
